File: API/models/SaleItem.js
'use strict';
const { Model } = require('sequelize');

module.exports = (sequelize, DataTypes) => {
  class SaleItem extends Model {
    static associate(models) {
      SaleItem.belongsTo(models.Sale, {
        foreignKey: 'sale_id',
        as: 'sale'
      });
      SaleItem.belongsTo(models.Material, {
        foreignKey: 'material_id',
        as: 'material'
      });
    }
  }

  SaleItem.init({
    sale_id: DataTypes.INTEGER,
    material_id: DataTypes.INTEGER,
    quantity: DataTypes.INTEGER,
    unit_price: DataTypes.DECIMAL
  }, {
    sequelize,
    modelName: 'SaleItem',
    tableName: 'sale_items',
    underscored: true,
    timestamps: true
  });

  return SaleItem;
};

File: StockFlowAPI/Models/SaleItem.cs
namespace StockFlowAPI.Models
{
    public class SaleItem
    {
        public int Id { get; set; }
        public int SaleId { get; set; }
        public int MaterialId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }

        public Sale Sale { get; set; } = null!;
        public Material Material { get; set; } = null!;
    }

}